include/openthread/dns.h

```diff
@@ -140,6 +141,93 @@ void otDnsInitTxtEntryIterator(otDnsTxtEntryIterator *aIterator, const uint8_t *
  */
 otError otDnsGetNextTxtEntry(otDnsTxtEntryIterator *aIterator, otDnsTxtEntry *aEntry);
 
+/**
+ * This function is called when a DNS-SD query subscribes a service or service instance.
+ *
+ * @param[in] aContext      A pointer to the application-specific context.
+ * @param[in] aFullName     The null-terminated full service name (e.g. "_ipps._tcp.default.service.arpa."), or
+ *                          full service instance name (e.g. "OpenThread._ipps._tcp.default.service.arpa.").
+ *
+ */
+typedef void (*otDnssdQuerySubscribe)(void *aContext, const char *aFullName);
```

Comment:
How about renaming to `otDnssdQuerySubscribeCallback`? (this make it more clear that it is a callback type and not a function)?